insights/parsers/vdo_status.py

```diff
@@ -103,14 +105,26 @@ def __get_dev_mapper__(self, vol):
 
         Raises:
             KeyError: If KEYs doesn't exist
+            ParseException: When input content is not available to parse
         """
 
+        # Check validation for vol name, return KeyError if not exists
         try:
-            dm_path = ('/dev/mapper/%s' % vol)
-            return self.data['VDOs'][vol]['VDO statistics'][dm_path]
+            self.data['VDOs'][vol]
         except:
-            err_path = "['VDOs'][%s]['VDO statistics'][%s]" % (vol, dm_path)
-            raise KeyError('No key(s) named: %s' % err_path)
+            err_path = "['VDOs'][%s]" % (vol)
+            raise KeyError('No key(s) named: %s in %s' % (vol, err_path))
+
+        # Parse next layer of ['VDO statistics'] when vol exists.
+        vs = self.data['VDOs'][vol]['VDO statistics']
+
+        # No device mapper path when 'VDO statistics' value is a string of 'not available'
+        if isinstance(vs, str):
```

Comment:
Better change to `if not isinstance(vs, dict):`

Reply:
yes, your suggestion is better.